crm-project/src/main/java/CRM/service/UserService.java

```diff
@@ -149,28 +151,54 @@ public List<User> getAllInBoard(long boardId) throws AccountNotFoundException {
      * @throws AccountNotFoundException if the user or board with the given id does not exist in the database
      * @throws IllegalArgumentException if the combination of the given user and board already exists in the database
      */
-    public void addUserToBoard(long userId, long boardId) throws AccountNotFoundException {
+    public List<User> updateUserToBoard(long userId, long boardId, long permissionId) throws AccountNotFoundException {
         User user;
         Board board;
         try {
             user = Validations.doesIdExists(userId, userRepository);
             board = Validations.doesIdExists(boardId, boardRepository);
-
         } catch (NoSuchElementException e) {
             throw new AccountNotFoundException(ExceptionMessage.ACCOUNT_DOES_NOT_EXISTS.toString());
         }
 
-        NotificationSetting notificationSetting = Validations.doesIdExists(2L, settingRepository);
+        if(user.equals(board.getCreatorUser())){
+            throw new IllegalArgumentException(ExceptionMessage.ADMIN_CANT_CHANGE_HIS_PERMISSION.toString());
+        }
 
-        UserPermission userPermission = new UserPermission();
-        userPermission.setId(0L);
-        userPermission.setUser(user);
-        userPermission.setPermission(Permission.USER);
+        UserPermission userPermissionInBoard = null;
+        Set<UserPermission> userPermissionsSet = board.getUsersPermissions();
 
-        board.addUserPermissionToBoard(userPermission);
+        for (UserPermission userInBoard : userPermissionsSet) {
+            if (userInBoard.getId().equals(userId)) {
+                userPermissionInBoard = userInBoard;
+                break;
+            }
+        }
```

Comment:
This whole code block has to be inside "board" entity. 
Just create a function there named "getUserPermissionById" and send it the required ID